src/tir/transforms/storage_rewrite.cc

```diff
@@ -1081,19 +1088,24 @@ struct BufferVarInfo {
       return element_dtype;
     }
 
+    // When there are scalar reads and no writes, access_dtype can be empty and we should avoid
+    // rewriting.
+    if (access_dtype.empty()) {
+      return element_dtype;
+    }
     DataType preferred_base_type = *base_access_dtype.begin();
 
-    // If there is only one vectorizable size used to access the
-    // buffer, and if that access size is compatible with the array
-    // size, then the buffer is vectorizable.  In the future, this
-    // could be improved to allow vectorized buffer access of size
-    // GCD(*lanes_used), if necessary.
```

Comment:
it seems taking gcd can cause regressions in some source codegen cases (e.g cuda). For example, when a buffer is accessed via `ramp(lanes=2)` and `ramp(lanes=4`), rewriting the buffer to float2 will disable the vectorization for float4.

Reply:
How about we only trigger rewriting if:
- there is vector write
- original dtype was float4
- there is only scalar read access

In a sense that comparing to original case, the only difference is that we allow scalar read from vector write

Reply:
That makes sense, I've updated rewriting. Now the rewriting only trigger if there is one vector access dtype with lanes same as the coeff scalar access.